File: vet/src/pojo/ProdutoQuimioterapico.java
package pojo;

/**
 * POJO 3 — ProdutoQuimioterapico
 * Subclasse de ProdutoVeterinario.
 * Representa medicamentos de síntese química (antibióticos, antiparasitários, etc.).
 */
public class ProdutoQuimioterapico extends ProdutoVeterinario {

    private String principioAtivo;    // Ex: "Ivermectina", "Amoxicilina"
    private double concentracao;      // Ex: 1.0 (em %)
    private String classeTerapeutica; // Ex: "Antibiótico", "Antiparasitário"
    private boolean retencaoCarencia; // Exige carência para abate/leite?

    public ProdutoQuimioterapico() {}

    public ProdutoQuimioterapico(int id, String nome, double preco, String fabricante,
                                 String registroMapa, String especieAlvo, String viaAdministracao,
                                 String principioAtivo, double concentracao,
                                 String classeTerapeutica, boolean retencaoCarencia) {
        super(id, nome, preco, fabricante, registroMapa, especieAlvo, viaAdministracao);
        this.principioAtivo    = principioAtivo;
        this.concentracao      = concentracao;
        this.classeTerapeutica = classeTerapeutica;
        this.retencaoCarencia  = retencaoCarencia;
    }

    // ── Getters / Setters ────────────────────────────────────────────────────

    public String getPrincipioAtivo()                        { return principioAtivo; }
    public void   setPrincipioAtivo(String principioAtivo)   { this.principioAtivo = principioAtivo; }

    public double getConcentracao()                          { return concentracao; }
    public void   setConcentracao(double concentracao)       { this.concentracao = concentracao; }

    public String getClasseTerapeutica()                             { return classeTerapeutica; }
    public void   setClasseTerapeutica(String classeTerapeutica)     { this.classeTerapeutica = classeTerapeutica; }

    public boolean isRetencaoCarencia()                          { return retencaoCarencia; }
    public void    setRetencaoCarencia(boolean retencaoCarencia) { this.retencaoCarencia = retencaoCarencia; }

    @Override
    public String toString() {
        return "ProdutoQuimioterapico{" +
               "id=" + getId() +
               ", nome='" + getNome() + '\'' +
               ", principioAtivo='" + principioAtivo + '\'' +
               ", concentracao=" + concentracao + "%" +
               ", classeTerapeutica='" + classeTerapeutica + '\'' +
               ", retencaoCarencia=" + retencaoCarencia + '}';
    }
}

File: vet/cliente/src/ProdutoQuimioterapico.cpp
#include "ProdutoQuimioterapico.hpp"

ProdutoQuimioterapico::ProdutoQuimioterapico(){};
ProdutoQuimioterapico::ProdutoQuimioterapico(int id, const string& nome, double preco, const string& fabricante, 
    const string& regM, const string& espA, const string& vAdm, 
    const string& principioAtivo, double concentracao, const string& classeTerapeutica, bool retencaoCarencia)
    : ProdutoVeterinario(id, nome, preco, fabricante, regM, espA, vAdm),
    principioAtivo(principioAtivo), concentracao(concentracao), classeTerapeutica(classeTerapeutica), retencaoCarencia(retencaoCarencia) {};

string ProdutoQuimioterapico::getPrincipioAtivo() const{
    return this->principioAtivo;
}

void ProdutoQuimioterapico::setPrincipioAtivo(const string& principioAtivo){
    this->principioAtivo = principioAtivo;
}

double ProdutoQuimioterapico::getConcentracao() const{
    return this->concentracao;
}

void ProdutoQuimioterapico::setConcentracao(double concentracao){
    this->concentracao = concentracao;
}

string ProdutoQuimioterapico::getClasseTerapeutica() const{
    return this->classeTerapeutica;
}

void ProdutoQuimioterapico::setClasseTerapeutica(const string& classeTerapeutica){
    this->classeTerapeutica = classeTerapeutica;
}

bool ProdutoQuimioterapico::ehRetencaoCarencia() const{
    return this->retencaoCarencia;
}

void ProdutoQuimioterapico::setRetencaoCarencia(bool retencaoCarencia){
    this->retencaoCarencia = retencaoCarencia;
}

string ProdutoQuimioterapico::toString() const{
    return "ProdutoQuimioterapico{id=" + to_string(this->getId()) + 
    ", nome='" + this->getNome() + 
    "', principioAtivo='" + this->principioAtivo + 
    "', concentracao=" + to_string(this->concentracao) + "%" + 
    ", classeTerapeutica='" + this->classeTerapeutica + 
    "', retencaoCarencia=" + (this->retencaoCarencia ? "true" : "false") + 
    "}";
}